R/FlattenDirectory.R

```diff
@@ -0,0 +1,23 @@
+#' Flatten a directory
+#'
+#' All files are copied from inDir to outDir. Nested directories are removed. For example, `src/module/home.R` becomes `R/module_home.R`
+#'
+#' @param strInDir a directory to flatten from
+#' @param strOutDir a directory to flatten to
+#' @param bOverwrite overwrite files in outDir?
+#'
+#' @importFrom purrr map_chr
+#'
+#' @return list of filenames added to outDir
+#' @export
+FlattenDirectory<-function(strInDir="src", strOutDir="R", bOverwrite=TRUE){
+    outFiles <-  map_chr(
+        list.files(strInDir, recursive=TRUE),
+        ~FlattenFile(
+            path=.x,
```

Comment:
What does the path=.x set for path?   I don't see a parameter x and am not familiar with .x?

Reply:
`.x` is the iteration parameter for the `purrr::map` family of fuctions. The code above is roughly equivalent to:
```
for(.x in list.files(strInDir, recursive=TRUE)) FlattenFile(path=.x)
```

Reply:
Thanks this is great to know